pystan/stanfit4model.pyx

```diff
@@ -551,19 +551,25 @@ cdef class StanFit4Model:
         If `permuted` is True, return dictionary with samples for each
         parameter (or other quantity) named in `pars`.
 
-        If `permuted` is False, an array is returned. The first dimension of
+        If `permuted` is False and `pars` is None, an array is returned. The first dimension of
         the array is for the iterations; the second for the number of chains;
         the third for the parameters. Vectors and arrays are expanded to one
         parameter (a scalar) per cell, with names indicating the third dimension.
         Parameters are listed in the same order as `model_pars` and `flatnames`.
+        
+        If `permuted` is False and `pars` is not None, return dictionary with samples for each
+        parameter (or other quantity) named in `pars`. The first dimension of
+        the sample array is for the iterations; the second for the number of chains;
+        the rest for the parameters. Parameters are listed in the same order as `pars`.
 
         """
         self._verify_has_samples()
         if inc_warmup is True and permuted is True:
             logging.warning("`inc_warmup` ignored when `permuted` is True.")
-        if dtypes is None and permuted is False:
-            logging.warning("`dtypes` ignored when `permuted` is False.")
+        if (dtypes is not None) and (permuted is False) and (pars is None):
```

Comment:
I think this is fine: ``if dtypes is not None and not permuted and pars is None``. https://docs.python.org/3/reference/expressions.html#operator-precedence

Reply:
Good point, I will clean it